src/JsonSchema/Constraints/ConstConstraint.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/*
+ * This file is part of the JsonSchema package.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace JsonSchema\Constraints;
+
+use JsonSchema\ConstraintError;
+use JsonSchema\Entity\JsonPointer;
+
+/**
+ * The ConstConstraint Constraints, validates an element against a constant value
+ *
+ * @author Martin Helmich <martin@helmich.me>
+ */
+class ConstConstraint extends Constraint
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function check(&$element, $schema = null, JsonPointer $path = null, $i = null)
+    {
+        // Only validate const if the attribute exists
+        if ($element instanceof UndefinedConstraint && (!isset($schema->required) || !$schema->required)) {
+            return;
+        }
+        $const = $schema->const;
+
+        $type = gettype($element);
+        $constType = gettype($const);
+
+        if ($this->factory->getConfig(self::CHECK_MODE_TYPE_CAST) && $type == 'array' && $constType == 'object') {
+            if ((object) $element == $const) {
+                return;
+            }
+        }
+
+        if ($type === gettype($const)) {
+            if ($type == 'object') {
+                if ($element == $const) {
```

Comment:
This does not correctly implement deep comparison of objects - it results in members being compared using `==`.

Could you please change this implementation to ensure type-strict comparison of object properties. You will probably need to iterate them; PHP lacks a suitable native comparison operator that can do it all in one go.

Reply:
Following that argumentation, the `enum` implementation (which was my inspiration for this class) should probably be changed as well -- it uses the same comparison.

Reply:
If enum is similarly broken, then yes - I agree with you that it needs fixing.